### ISSUE TITLE
params added by `phx-click` do not follow the same naming convention as `phx-value-*`

### ISSUE THREAD

**Issue by Qqwy** [phoenixframework/phoenix_live_view#819]
### Situation

When clicking on the following element:

```
<button phx-click="some-event" phx-value-button-name="My Button">
  My Beautiful Button
</button>
```

### Actual behavior

The parameters contained in the event dispatched by LiveView are as follows

```
%{
  "altKey" => false,
  "ctrlKey" => false,
  "detail" => 1,
  "button-name" => "My Button",
  "metaKey" => false,
  "offsetX" => 340,
  "offsetY" => 18,
  "pageX" => 756,
  "pageY" => 205,
  "screenX" => 2675,
  "screenY" => 302,
  "shiftKey" => false,
  "x" => 756,
  "y" => 205
}
```

### Expected behavior

I would expect the keys injected by `phx-click` and the keys added by `phx-value-*` to follow the same naming convention:

- Either `button-name` being converted to (lower) camelCase `buttonName`;
- or the special meta-keys that are being added by `phx-click` (`altKey`, `ctrlKey`, etc.) being converted into to dash-style (`alt-key`, `ctrl-key`, etc.).

**Comment by josevalim** [comment]
I am honestly torn. Rewriting the browser ones means users need to keep to relearn the browser names. At the same time, if we were to rewrite button-name, I would rather call it button_name. I am also not a big fan of automatically sending all click events automatically either. I wish they were opt-in. Maybe we could do “phx-value-screen-x” and if not value is supplied, we read it for you.

**Comment by Qqwy** [comment]
Another option w.r.t. the naming might be to only use _one_ special key of the parameters for the click events which then contains the map with all values, in which case the difference in naming might be less problematic.

About sending all click events: There are many cases in which they are not required, but then again cases where they are rather important exist as well. I don't link the approach of altering the behaviour of `phx-value` based on if a value has passed in. I think adding an extra attribute to toggle it (or maybe a special variant of `phx-click`, say `phx-click-advanced`) might be easier to understand for the user.

**Comment by josevalim** [comment]
I like the idea of putting the client specific ones under a specific key. Or maybe we add some special values: “phx-value-screen=$screen-info”

**Comment by Qqwy** [comment]
The problem with special values is of course that we need to handle the case where someone actually _does_ want to store e.g. the actual string `$screen-info`. Resolving this ambiguity means that we need some alternate syntax that is only used for 'special values'.

Another possibility is to add a setting to the instantiation of the `LiveSocket`-object in the JS file, making it configurable (only) for the whole app as a whole which client values are sent with each (mouse-related) request.

**Comment by josevalim** [comment]
This will be fixed by #868.
